### PR TITLE
[Merged by Bors] - feat: add some symm attributes throughout the library

### PR DESCRIPTION
**Pull request by alexjbest**
Also add a couple of refl and trans attributes 

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/RingTheory/IsTensorProduct.lean [hunk 1]
```diff
@@ -382,6 +382,7 @@ class Algebra.IsPushout : Prop where
 
 variable {R S R' S'}
 
+@[symm]
 theorem Algebra.IsPushout.symm (h : Algebra.IsPushout R S R' S') : Algebra.IsPushout R R' S S' := by
```
Comment (eric-wieser) [inline Mathlib/RingTheory/IsTensorProduct.lean:386]: Does `symm` really work on this one?

File: Mathlib/SetTheory/ZFC/Basic.lean [hunk 2]
```diff
@@ -557,6 +557,7 @@ def Resp.Equiv {n} (a b : Resp n) : Prop :=
   Arity.Equiv a.1 b.1
 #align pSet.resp.equiv PSet.Resp.Equiv
 
+@[refl]
```
Comment (eric-wieser) [inline Mathlib/SetTheory/ZFC/Basic.lean:560]: This is a weird way to spell `symm`! Please mention there's a `refl` attribute too in the PR description.

File: Mathlib/SetTheory/ZFC/Basic.lean [hunk 3]
```diff
@@ -568,10 +569,12 @@ protected theorem Resp.Equiv.euc :
     @Resp.Equiv.euc n (a.f x) (b.f y) (c.f y) (hab _ _ h) (hcb _ _ <| PSet.Equiv.refl y)
 #align pSet.resp.equiv.euc PSet.Resp.Equiv.euc
 
+@[symm]
 protected theorem Resp.Equiv.symm {n} {a b : Resp n} : Resp.Equiv a b → Resp.Equiv b a :=
   (Resp.Equiv.refl b).euc
 #align pSet.resp.equiv.symm PSet.Resp.Equiv.symm
 
+@[trans]
```
Comment (eric-wieser) [inline Mathlib/SetTheory/ZFC/Basic.lean:577]: And `trans` too

**Review by eric-wieser — APPROVED** [review]
bors d+

File: Mathlib/RingTheory/IsTensorProduct.lean (hunk 1 above)
Comment (alexjbest) [reply Mathlib/RingTheory/IsTensorProduct.lean:386]: Yes, I checked the ones that I was unsure about, it seems symm is more versatile than I understood it to be

**Pull request closed without merging**